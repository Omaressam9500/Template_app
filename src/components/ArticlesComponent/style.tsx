import { StyleSheet } from 'react-native';
import { Pixel } from '../../constants/styleConstatnts';
import { useTheme } from '../../context/ThemeProvider';
export const styles = () => {
    const { theme } = useTheme();

    return StyleSheet.create({

        card: {
            backgroundColor: theme.backgroundColor,
            borderRadius: 8,
            shadowColor: '#000',
            shadowOffset: { width: 0, height: 2 },
            shadowOpacity: 0.2,
            shadowRadius: 4,
            elevation: 3,
            marginBottom: 20,
        },
        imageStyle: {
            width: '100%',
            height: 200,
            borderTopLeftRadius: 8,
            borderTopRightRadius: 8,
    
        },
        content: {
            padding: 10,
        },
        titleStyle: {
            fontSize: 18,
            fontWeight: 'bold',
            marginBottom: 5,
            alignSelf: 'flex-start',
            color:theme.textColor
        },
        descriptionStyle: {
            fontSize: 16,
            alignSelf: 'flex-start',
            color:theme.textColor
        },
    })
}