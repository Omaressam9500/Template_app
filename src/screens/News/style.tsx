import { StyleSheet } from 'react-native';
import { Pixel } from '../../constants/styleConstatnts';
import { useTheme } from '../../context/ThemeProvider';
export const styles = () => {
    const { theme, } = useTheme();

    return StyleSheet.create({

        mainContainer: {
            paddingHorizontal: Pixel(16),
            backgroundColor: theme.screenBackgroundColor


        },
        listContentStyle: {
            marginTop: Pixel(23),
            paddingBottom: Pixel(140),
            paddingTop: 0,
        },
        emptyView:{
            flex:1,
            alignItems:'center',
            justifyContent:'center',
            height:Pixel(150)
        }
    })
}