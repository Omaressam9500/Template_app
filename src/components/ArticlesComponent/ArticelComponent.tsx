import React, { FC } from "react";
import { StyleSheet, Text, View, Image, TouchableOpacity, } from "react-native";

import { useNavigation } from "@react-navigation/native";
import { Pixel } from "../../constants/styleConstatnts";
import { ImageComponent } from "../ImageComponent";
import { Screens } from "../../constants/Screens";
import { styles } from "./style";

interface IArticelComponent {
    post: IArticle
}
export const ArticelComponent: FC<IArticelComponent> = ({
    post
}) => {
    const { navigate } = useNavigation();
    const { author, urlToImage, title } = post;
    const { card, imageStyle, content, titleStyle, descriptionStyle } = styles()
    return (
        <TouchableOpacity
        testID="CardTestId"
         style={card}
            onPress={() => {
                navigate(Screens.ARTICLES_DETAILS, {
                    post: post
                })
            }}>
            <ImageComponent
                url={urlToImage}
                style={imageStyle}
                resizeMode={"cover"}
            />

            <View style={content}>
                <Text testID={"AuthorTestId"}
                    style={titleStyle} numberOfLines={2} ellipsizeMode="tail">{author}</Text>
                <Text testID={"TitleTestId"}
                    style={descriptionStyle} numberOfLines={2} ellipsizeMode="tail">{title}</Text>

            </View>

        </TouchableOpacity>


    )
}
