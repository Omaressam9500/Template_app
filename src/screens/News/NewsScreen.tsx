import { t } from "i18next";
import React, { useState, useEffect, useContext } from "react";
import { View, StyleSheet, TextInput, FlatList, Image, Text, ActivityIndicator, RefreshControl } from 'react-native'
import { useQueryClient } from "react-query";
import { getArticles } from "../../api-hooks/getArticles";
import { ArticelComponent } from "../../components/ArticlesComponent/ArticelComponent";
import SearchBar from "../../components/SearchBarComponent/SearchBar";
import { Container } from "../../constants/Containers";
import { useTheme } from '../../context/ThemeProvider';
import { styles } from "./style";


export const NewsScreen = () => {
    const { mainContainer, listContentStyle,emptyView } = styles();
    const [searchName, setSearchName] = useState('tesla')
    const [articlesList, setArticlesList] = useState([])
    const { theme } = useTheme();

    const { isFetching: isArticlesFetching,
        refetch: refetchArticles
    } = getArticles(searchName, (data) => {
        setArticlesList(data?.data?.articles)
    })
    return (
        <Container style={mainContainer} >
            <SearchBar
                handleSearchValue={(search) => {
                    setSearchName(search)
                }}
            />
            <FlatList
                data={articlesList}
                legacyImplementation
                testID="articlesListId"
                renderItem={({ item, index }) => {
                    {
                        return (
                            <ArticelComponent
                                post={item}
                            />

                        )
                    }
                }}
                refreshControl={
                    <RefreshControl refreshing={isArticlesFetching} onRefresh={refetchArticles}
                        tintColor={theme.black} />
                }
                ListEmptyComponent={() => {
                    return (
                        <View style={emptyView}>
                            {!isArticlesFetching &&
                                <Text testID="loading-message">{t('NoData')}</Text> 


                            }
                        </View>
                    )
                }
                }

                contentContainerStyle={listContentStyle}
                showsHorizontalScrollIndicator={false}
                showsVerticalScrollIndicator={false}
            />
        </Container>
    )


}
