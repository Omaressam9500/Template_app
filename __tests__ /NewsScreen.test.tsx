import React from 'react';
import { it, expect, describe, jest } from '@jest/globals';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { FlatList } from 'react-native';
import { QueryClient, QueryClientProvider } from 'react-query';
import { ArticelComponent } from '../src/components/ArticlesComponent/ArticelComponent';
import { NewsScreen } from '../src/screens/News/NewsScreen';
jest.mock('@react-native-async-storage/async-storage', () =>
    require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);
describe('NewsScreen', () => {

    const queryClient = new QueryClient()
    const TestWrapper = ({ children }) => (
        <QueryClientProvider client={queryClient}>
            {children}
        </QueryClientProvider>
    );
    it('Render ComponentCorrect', () => {

        const tree = render(<NewsScreen />, { wrapper: TestWrapper })
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
   
});
