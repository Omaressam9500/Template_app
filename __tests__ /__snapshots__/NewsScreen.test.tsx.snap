// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsScreen Render ComponentCorrect 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F9F9FF",
        "flex": 1,
        "paddingTop": undefined,
      },
      {
        "backgroundColor": "#F9F9FF",
        "paddingHorizontal": 27,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "marginTop": 25,
          "paddingBottom": 17,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#F9F9FF",
          "borderColor": "#000000",
          "borderRadius": 94,
          "borderWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "fontSize": 23,
          "height": 82,
          "marginRight": 12,
          "paddingLeft": 25,
          "paddingRight": 17,
        }
      }
    >
      <TextInput
        focusable={true}
        onChangeText={[Function]}
        placeholderTextColor="#919C9C"
        returnKeyType="search"
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontSize": 23,
            "textAlign": "left",
          }
        }
        testID="input"
        value=""
      />
    </View>
  </View>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    contentContainerStyle={
      {
        "marginTop": 39,
        "paddingBottom": 239,
        "paddingTop": 0,
      }
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    legacyImplementation={true}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={true}
        tintColor="#000000"
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    testID="articlesListId"
    viewabilityConfigCallbackPairs={[]}
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "height": 256,
            "justifyContent": "center",
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
